pkg/driver/mysql.go

```diff
@@ -0,0 +1,117 @@
+package driver
+
+import (
+	"context"
+	"database/sql/driver"
+	"github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+// MySQLDriver extends mysql.MySQLDriver with auto-SETting Galera cluster options.
+type MySQLDriver struct {
+	mysql.MySQLDriver
+}
+
+// Open implements the driver.Driver interface.
+func (md MySQLDriver) Open(name string) (driver.Conn, error) {
+	connector, err := md.OpenConnector(name)
+	if err != nil {
+		return nil, err
+	}
+
+	return connector.Connect(context.Background())
+}
+
+// OpenConnector implements the driver.DriverContext interface.
+func (md MySQLDriver) OpenConnector(name string) (driver.Connector, error) {
+	var wsrepSyncWait int64
+
+	if config, err := mysql.ParseDSN(name); err == nil {
+		if s, ok := config.Params["wsrep_sync_wait"]; ok {
+			if i, err := strconv.ParseInt(s, 10, 64); err == nil {
+				wsrepSyncWait = i
+				delete(config.Params, "wsrep_sync_wait")
```

Comment:
Storing `wsrep_sync_wait` inside the `config.Params` feels a bit like a cheat, especially as it will be deleted at this point. Maybe add at least a short comment explaining this?